jax/experimental/sparse_ops.py

```diff
@@ -440,3 +447,129 @@ def _coo_matmat_gpu_translation_rule(c, data, row, col, B, *, shape, transpose):
 if cusparse and cusparse.is_supported:
   xla.backend_specific_translations['gpu'][
       coo_matmat_p] = _coo_matmat_gpu_translation_rule
+
+#----------------------------------------------------------------------
+# Sparse objects (APIs subject to change)
+class JAXSparse:
+  """Base class for high-level JAX sparse objects."""
+  _data: Tuple[jnp.ndarray, jnp.ndarray, jnp.ndarray]
```

Comment:
nit: This feels a little out of place to me in a baseclass. It is not obvious that there will always be exactly three arrays backing each sparse matrix. I would lean towards leaving `_data` and `__init__` up to subclasses to implement. I guess it works for now, though.

Reply:
Thanks for the suggestion - I rearranged things a bit, I think it's cleaner now.